presto-raptor/src/main/java/com/facebook/presto/raptor/storage/ShardStats.java

```diff
@@ -0,0 +1,249 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.storage;
+
+import com.facebook.presto.orc.BooleanVector;
+import com.facebook.presto.orc.DoubleVector;
+import com.facebook.presto.orc.LongVector;
+import com.facebook.presto.orc.OrcPredicate;
+import com.facebook.presto.orc.OrcReader;
+import com.facebook.presto.orc.OrcRecordReader;
+import com.facebook.presto.orc.SliceVector;
+import com.facebook.presto.raptor.metadata.ColumnStats;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.BigintType;
+import com.facebook.presto.spi.type.BooleanType;
+import com.facebook.presto.spi.type.DateType;
+import com.facebook.presto.spi.type.DoubleType;
+import com.facebook.presto.spi.type.TimestampType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.VarcharType;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.slice.Slice;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.raptor.RaptorErrorCode.RAPTOR_ERROR;
+import static java.lang.Double.isInfinite;
+import static java.lang.Double.isNaN;
+import static org.joda.time.DateTimeZone.UTC;
+
+public final class ShardStats
+{
+    /**
+     * Maximum length of a binary value stored in an index.
+     */
+    public static final int MAX_BINARY_INDEX_SIZE = 100;
+
+    private ShardStats() {}
+
+    public static Slice truncateIndexValue(Slice slice)
+    {
+        if (slice.length() > MAX_BINARY_INDEX_SIZE) {
+            return slice.slice(0, MAX_BINARY_INDEX_SIZE);
+        }
+        return slice;
+    }
+
+    public static ColumnStats computeColumnStats(OrcReader orcReader, long columnId, Type type)
+            throws IOException
+    {
+        int columnIndex = columnIndex(orcReader.getColumnNames(), columnId);
+        Set<Integer> includedColumns = ImmutableSet.of(columnIndex);
+        OrcRecordReader reader = orcReader.createRecordReader(includedColumns, OrcPredicate.TRUE, UTC);
+
+        if (type.equals(BooleanType.BOOLEAN)) {
+            return indexBoolean(reader, columnIndex, columnId);
+        }
+        if (type.equals(BigintType.BIGINT) ||
+                type.equals(DateType.DATE) ||
+                type.equals(TimestampType.TIMESTAMP)) {
+            return indexLong(reader, columnIndex, columnId);
+        }
+        if (type.equals(DoubleType.DOUBLE)) {
+            return indexDouble(reader, columnIndex, columnId);
+        }
+        if (type.equals(VarcharType.VARCHAR)) {
+            return indexString(reader, columnIndex, columnId);
+        }
+        return null;
+    }
+
+    private static int columnIndex(List<String> columnNames, long columnId)
+    {
+        int index = columnNames.indexOf(String.valueOf(columnId));
```

Comment:
We should not have this logic here, instead get a list of columns from the OrcStorageManger